src/geocoding/geocoder.py

```
@@ -0,0 +1,722 @@
+                    
+                    self.circuit_breaker.record_success()
+                    return result
+                    
+        except asyncio.TimeoutError:
+            self.circuit_breaker.record_failure()
+            return GeocodeResult(
+                status=GeocodeStatus.FAILED,
+                provider=GeocodeProvider.NOMINATIM,
+                response_time_ms=(time.time() - start_time) * 1000
+            )
+        except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)